framework/src/headers/ip/v6/nf_macros.rs

```diff
@@ -0,0 +1,91 @@
+/*
+  Helper for inserting an SRH into a V6 Header and having it generate concrete
+  types for a range of segment lengths.
+
+  Macro parameters include:
+
+  o $segments - a vector of segments to insert as part of creating an SRH.
+  o $pkt - the current packet being operated on within a network function.
+  o $sr_info - SRH information, e.g. segments_left, tag, etc... that you want to
+               set upon insertion/creation.
+  o $prev_hdr_type - previous header type for SegmentRoutingHeader construction.
+  o [$len => $utype] - number-to-type pairs to allow for static writes of
+                       runtime numeric values, e.g. [1 => U1, 2 => U2]
+*/
+#[macro_export]
+macro_rules! srh_insert {
+    ($segments:expr, $pkt:expr, $sr_info:expr, $prev_hdr_type:ty, [$($len:expr => $utype:ty),*]) => {
+        match $segments.len() {
+            $(
+                $len => {
+                    // TODO: Eventually maybe make this a hashmap, iterator (k,v), struct
+                    let mut srh = <SegmentRoutingHeader<$prev_hdr_type, $utype>>::new_from_tuple(
+                        $sr_info,
+                        *GenericArray::<_, $utype>::from_slice(&$segments[..]),
+                    );
+
+                    if let Ok(()) = $pkt.insert_header(&srh) {
+                        let (_next_header, segments_left, _flags, _tags) = $sr_info;
+
+                        let payload_len = $pkt.get_header().payload_len();
+                        let mut v6h = $pkt.get_mut_header();
+                        v6h.set_next_header(NextHeader::Routing);
+                        v6h.set_dst(Segment::from($segments[segments_left as usize]));
+                        v6h.set_payload_len(payload_len + srh.offset() as u16);
```

Comment:
Should this return the new `dst` addr too, since `swap` does?

Reply:
OK, guess it doesn't have to, since it goes ahead and swaps it into the v6 header. Same with potentially returning the size diff, since the payload_len is already set.